packages/ejson_generator/lib/src/generator.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 MongoDB, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:build/build.dart';
+import 'package:ejson_analyzer/ejson_analyzer.dart';
+import 'package:ejson_generator/ejson_generator.dart';
+import 'package:source_gen/source_gen.dart';
+
+extension on EJsonError {
+  String get message => switch (this) {
+        EJsonError.tooManyAnnotatedConstructors => 'Too many annotated constructors',
+        EJsonError.missingGetter => 'Missing getter',
+        EJsonError.mismatchedGetterType => 'Mismatched getter type',
+        EJsonError.tooManyConstructorsOnAnnotatedClass => 'Too many constructors on annotated class',
+        EJsonError.noExplicitConstructor => 'No explicit constructor',
+      };
+
+  Never raise() {
+    throw EJsonSourceError._(this);
+  }
```

Comment:
Rather than have extensions on the enum, isn't it better to use the cool new enums that shipped with Flutter 3?

Reply:
Fixed